### PR TITLE
MAO: add progressing/failing verification in available e2e check

### PR DESCRIPTION
**Pull request by michaelgugino**

### CODE REVIEW

File: pkg/e2e/operators/cluster-autoscaler-operator.go [hunk 1]
```diff
@@ -62,12 +62,25 @@ var _ = g.Describe("[Feature:Operators] Cluster autoscaler cluster operator shou
 				glog.Errorf("error querying api for OperatorStatus object: %v, retrying...", err)
 				return false, nil
 			}
+			ready := false
 			if available := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorAvailable); available != nil {
 				if available.Status == osconfigv1.ConditionTrue {
-					return true, nil
+					ready = true
+				} else {
+					ready = false
```
Comment (ingvagabund) [inline pkg/e2e/operators/cluster-autoscaler-operator.go]: you might just do `return false, nil` here

File: pkg/e2e/operators/cluster-autoscaler-operator.go [hunk 2]
```diff
@@ -62,12 +62,25 @@ var _ = g.Describe("[Feature:Operators] Cluster autoscaler cluster operator shou
...
 				}
 			}
-			return false, nil
+			if progressing := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorProgressing); progressing != nil {
+				if progressing.Status == osconfigv1.ConditionTrue {
+					ready = false
```
Comment (ingvagabund) [inline pkg/e2e/operators/cluster-autoscaler-operator.go]: the same here: `return false, nil` here

File: pkg/e2e/operators/cluster-autoscaler-operator.go [hunk 3]
```diff
@@ -62,12 +62,25 @@ var _ = g.Describe("[Feature:Operators] Cluster autoscaler cluster operator shou
...
+				}
+			}
+			if failing := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorFailing); failing != nil {
+				if failing.Status == osconfigv1.ConditionTrue {
+					ready = false
```
Comment (ingvagabund) [inline pkg/e2e/operators/cluster-autoscaler-operator.go]: the same here: `return false, nil` here

File: pkg/e2e/operators/cluster-autoscaler-operator.go [hunk 4]
```diff
@@ -62,12 +62,25 @@ var _ = g.Describe("[Feature:Operators] Cluster autoscaler cluster operator shou
...
+				}
+			}
+			return ready, nil
```
Comment (ingvagabund) [inline pkg/e2e/operators/cluster-autoscaler-operator.go]: and here return `return true, nil` and remove `ready` var completely.

Comment (michaelgugino) [reply pkg/e2e/operators/cluster-autoscaler-operator.go]: @ingvagabund done

**Pull request merged**